DDDEastAnglia/Areas/Admin/Controllers/SponsorController.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.IO;
+using System.Linq;
+using System.Web;
+using System.Web.Mvc;
+using DDDEastAnglia.Areas.Admin.Models;
+using DDDEastAnglia.DataAccess;
+using DDDEastAnglia.Helpers;
+using DDDEastAnglia.Models;
+
+namespace DDDEastAnglia.Areas.Admin.Controllers
+{
+    public class SponsorController : Controller
+    {
+        private readonly ISponsorRepository sponsorRepository;
+        private readonly ISponsorSorter sponsorSorter;
+
+        public SponsorController(ISponsorRepository sponsorRepository, ISponsorSorter sponsorSorter)
```

Comment:
I originally tried that, but this controller still needs the repository (for the create and delete methods that don't care about sorting), so it seems a bit pointless.